### PR TITLE
Libdl: implement dllist on FreeBSD

### PR DESCRIPTION
**Pull request by iblislin**
DL_ITERATE_PHDR(3):
https://www.freebsd.org/cgi/man.cgi?query=dl_iterate_phdr&apropos=0&sektion=3&manpath=FreeBSD+12-current&arch=default&format=html

### CODE REVIEW

File: base/libdl.jl [hunk 1]
```diff
@@ -205,6 +205,43 @@ if is_linux()
     end
 end # linux-only
 
+if Sys.KERNEL == :FreeBSD
```
Comment (ararslan) [inline base/libdl.jl]: Is this entirely unique to FreeBSD, or will it apply to Dragonfly and other BSDs? If the latter, this should be `if is_bsd() && !is_apple()`.

Comment (iblis17) [inline base/libdl.jl]: I checked the manual,  found that they share the same c struct in the first 4 field, but not sure the behaviour is consistent.

I will install VMs and try  later.

NetBSD: http://netbsd.gw.com/cgi-bin/man-cgi?dl_iterate_phdr+3+NetBSD-7.0
OpenBSD: http://man.openbsd.org/OpenBSD-current/man3/dl_iterate_phdr.3
DragonflyBSD: https://www.dragonflybsd.org/cgi/web-man?command=dl_iterate_phdr&section=3

Comment (iblis17) [inline base/libdl.jl]: The behavior is same on NetBSD and Dragonfly. But OpenBSD will get compilation error with
`#include <link.h>`. (`link.h` is described on OpenBSD's manual, seems there are some syntax error in the header file)

I decide to use `f is_bsd() && !is_apple()`.

Comment (iblis17) [inline base/libdl.jl]: On more question: is sutable to use `@static` macro here? I just found that the `if is_linux` section in this file do not use `@static`. what is the reason?

Comment (yuyichao) [inline base/libdl.jl]: You don't need `@static` at the top level 99% of the time since that's already what the branch is doing.

**Review by ararslan — APPROVED** [review]

**Pull request merged**